src/actions/facebookcustomermatch/lib/executor.ts

```diff
@@ -0,0 +1,448 @@
+  private operationType: string
+  private adAccountId: string
+  private customAudienceId: string | undefined = ""
+  private customAudienceName: string | undefined = ""
+  private customAudienceDescription: string | undefined = ""
+
+  constructor(actionRequest: Hub.ActionRequest, accessToken: string) {
+    this.actionRequest = actionRequest
+    this.sessionId =  Date.now() // a unique id used to associate multiple requests with one custom audience API action
+    this.facebookAPI = new FacebookCustomerMatchApi(accessToken)
+    this.shouldHash = actionRequest.formParams.should_hash === "do_no_hashing" ? false : true
+    const operationType = actionRequest.formParams.choose_create_update_replace
+    if(!operationType) {
+      throw new Error("Cannot execute action without choosing an operation type.")
+    }
+    this.operationType = operationType
+
+    if(!actionRequest.formParams.choose_business || !actionRequest.formParams.choose_ad_account) {
+      throw new Error("Cannot execute action without business id or ad account id.")
+    }
+    if(!actionRequest.formParams.choose_custom_audience && (
+      operationType === "update_audience" || operationType === "replace_audience"
+    )) {
+      throw new Error("Cannot update or replace without a custom audience id.")
+    }
+
+    this.adAccountId = actionRequest.formParams.choose_ad_account
+    this.customAudienceId = actionRequest.formParams.choose_custom_audience
+    this.customAudienceName = actionRequest.formParams.create_audience_name
+    this.customAudienceDescription = actionRequest.formParams.create_audience_description
+  }
+
+  private fieldMapping : FieldMapping[] = [
+    {
+      lookMLFieldName: "Email",
+      fallbackRegex: /email/i,
+      userField: "email",
+      normalizationFunction: this.normalize
+    },
+    {
+      lookMLFieldName: "Phone",
+      fallbackRegex: /phone/i,
+      userField: "phone",
+      normalizationFunction: removeNonRomanAlphaNumeric
+    },
+    {
+      lookMLFieldName: "Gender",
+      fallbackRegex: /gender/i,
+      userField: "gender",
+      normalizationFunction: (value: string) => genderNameToCode(this.normalize(value))
+    },
+    {
+      lookMLFieldName: "BirthYear",
+      fallbackRegex: /year/i,
+      userField: "birthYear",
+      normalizationFunction: (value: string) => formatFullDate(this.normalize(value))
+    },
+    {
+      lookMLFieldName: "BirthMonth",
+      fallbackRegex: /month/i,
+      userField: "birthMonth",
+      normalizationFunction: (value: string) => getMonth(this.normalize(value))
+    },
+    {
+      lookMLFieldName: "BirthDay",
+      fallbackRegex: /day/i,
+      userField: "birthDay",
+      normalizationFunction: (value: string) => getDayOfMonth(this.normalize(value))
+    },
+    {
+      lookMLFieldName: "LastName",
+      fallbackRegex: /last/i,
+      userField: "lastName",
+      normalizationFunction: this.normalize
+    },
+    {
+      lookMLFieldName: "FirstName",
```

Comment:
LookML objects are typically snake_case not camelcase e.g. `first_name` not `FirstName`. Can we update the lookMLFieldName values respectively? I think this will get a better hit rate